File: entsoe-client/src/main/java/org/example/model/common/AuctionType.java
package org.example.model.common;

import java.util.Arrays;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum AuctionType {
  IMPLICIT("A01", "Implicit"),
  EXPLICIT("A02", "Explicit");

  private final String code;
  private final String description;

  public static AuctionType fromCode(String code) {
    return Arrays.stream(values())
        .filter(type -> type.code.equals(code))
        .findFirst()
        .orElseThrow(() -> new IllegalArgumentException("Unknown AuctionType code: " + code));
  }
}

File: entsoe_client/src/entsoe_client/model/common/auction_type.py
from enum import Enum
from typing import Self

from src.entsoe_client.exceptions.unknown_auction_type_error import (
    UnknownAuctionTypeError,
)


class AuctionType(Enum):
    IMPLICIT = ("A01", "Implicit")
    EXPLICIT = ("A02", "Explicit")

    def __init__(self, code: str, description: str):
        self.code = code
        self.description = description

    @classmethod
    def from_code(cls, code: str) -> Self:
        for member in cls:
            if member.code == code:
                return member
        raise UnknownAuctionTypeError
